functional/cypress-tests/cypress/support/widgets.js

```diff
@@ -39,10 +39,14 @@ Cypress.Commands.add('confirmAddWidget', () => {
         .click();
 });
 
-Cypress.Commands.add('removeWidget', () => {
-    cy.get('[data-cy="more-menu-button"]')
+Cypress.Commands.add('removeWidget', name => {
+    cy.contains(name)
+        .parents('.MuiCardHeader-root')
+        .find('[data-cy="more-menu-button"]')
         .click();
-    cy.get('[data-cy="widget-delete"]')
+    cy.get('div[id="more-menu"]')
+        .not('[aria-hidden="true"]')
+        .find('[data-cy="widget-delete"]')
```

Comment:
What is the reason for elongation of the command?

Reply:
The reason is to get only one [data-cy="widget-delete"] element that 'belongs' to widget we want to delete. Previous command would get e.g. 3 [data-cy="widget-delete"] elements if there are 3 widgets.